ui-hello-react/src/components/ColorPicker.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { editDocument } from 'application';
+import { Color, selection } from 'scenegraph';
+import './ColorPicker.css';
+
+import Slider from './Slider';
+
+const ColorPicker = () => {
+  const _sldRed = useRef(null);
+  const _sldGreen = useRef(null);
+  const _sldBlue = useRef(null);
+  const _sldAlpha = useRef(null);
+  const _textRed = useRef(null);
+  const _textGreen = useRef(null);
+  const _textBlue = useRef(null);
+  const _textAlpha = useRef(null);
+
+  const [red, setRed] = useState(255);
+  const [green, setGreen] = useState(0);
+  const [blue, setBlue] = useState(0);
+  const [alpha, setAlpha] = useState(1);
+
+  const [render, setRender] = useState(false);
+
+  useEffect(() => {
+    if (render) {
+      updateDocumentColor();
+    } else {
+      setRender(true);
+    }
+  }, [red, green, blue, alpha]);
```

Comment:
Decided to use a state variable as a workaround for updating the XD document (as mentioned in [this forum post](https://forums.creativeclouddeveloper.com/t/useeffect-hook-causes-a-panel-plugin-edits-must-be-initiated-from-a-supported-ui-event-error/2720/3)). 

For the context of this sample, the color state variables won't be rendered after the initial render unless they are triggered from a panel UI event. Let me know if this works for the sake of the sample.